Libraries/Game/game.cpp

```diff
@@ -0,0 +1,132 @@
+#include "game.h"
+
+bool Game::init(int size, int sp){
+    dim.x = size;
+    dim.y = size/2;
+    setLabyrinth(dim);
+    setSpeed(sp);
+    return true;
+}
+
+int Game::run(){
+
+    initscr();
+
+    nodelay(stdscr, TRUE);
+    set_escdelay(0);
+
+    Point screenSize;
+    getmaxyx(stdscr, screenSize.y, screenSize.x);
+
+    Point p, g;
+    p.x = 1;
+    p.y = 1;
+    int dir = 1, len = 5;
+    initSnake(p, dir, len);
+
+    curs_set(0); //"Убиваем" курсор
+    //Включаем режим удобной работы с функциональными клавишами, другими словами KEY_UP и KEY_DOWN без этого не работали бы
+    keypad(stdscr, true); 
+    bool flag = false;
+
+    Ball ball(dim);
+    ball.generateBall(labyrinth);
+
+    Point* change[2];
+    change[0] = new Point [1];
+    change[1] = new Point [1];
+    while ( !flag )
+    {
+
+    	int ch = getch();
+
+    	switch (ch){
+    		case ERR:
+    			break;
+    		case KEY_UP:
+    			snake.setDirection(-2);
+    			break;
+    		case KEY_DOWN:
+    			snake.setDirection(2);
+    			break;
+    		case KEY_LEFT:
+    			snake.setDirection(-1);
+    			break;
+    		case KEY_RIGHT:
+    			snake.setDirection(1);
+    			break;
+            case 'q':
+                flag = true;
+                break;
+
+    	}
+        if (flag) break;
+    	flag = snake.move(labyrinth, &ball, change);
+        updateLabyrinth(change, 1);
+    	clear();
+        displayLabyrinth();
+    	for(auto it = snake.snakeBody.begin(); it != snake.snakeBody.end(); it++){
+    		mvaddch((*it).y, (*it).x, '*');
+    	}
+        g = ball.getCoords();
+        mvaddch(g.y, g.x, '*');
+    	flushinp();
+    	mSleep(speed);
+    	refresh();
+        if (flag) break;
```

Comment:
Duplicate of 63. Is it necessary?

Reply:
it is necessary because right after the first use it gets assigned to the snake.move() result
And if the snake 'dies' during the move we still exit the game loop